CIPs/cip-137.md

```diff
@@ -0,0 +1,203 @@
+---
+cip: 137
+title: Ceramic API
+author: Danny Browning <dbrowning@3box.io>, Nathaniel Cook <nathaniel@3box.io>, Aaron Goldman <aaron@3box.io>, Joel Thorstensson <joel@3box.io>
+discussions-to: https://forum.ceramic.network/t/cip-137-ceramic-api/
+status: Draft
+category: Interface
+created: 2023-05-28
+edited: 2023-06-01
+requires: 124
+---
+
+
+## Simple Summary
+<!--Provide a simplified and layman-accessible explanation of the CIP.-->
+An API for reading and writing events on Ceramic event streams.
+
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+This CIP introduces a JSON-RPC interface that can be used to subscribe to a number of streams inside of an interest range, and write new events to these streams. The API exposes two main ways of retrieving events. A simple method that uses the nodes local ordering of events to retrieve events by simply keeping track of a single vector client side. More advanced users can use a simplified version of the [Recon](https://cips.ceramic.network/CIPs/cip-124) protocol to retrieve events in their logical order.
```

Comment:
why JSON-RPC over say GraphQL

Reply:
Sure, happy to change this if we have a good reason for it! Let's continue this discussion in the forum